support-frontend/assets/helpers/tracking/quantumMetric.ts

```diff
@@ -264,46 +269,52 @@ function sendEventContributionCartValue(
 		return;
 	}
 
-	void canRunQuantumMetric().then((canRun) => {
-		if (canRun) {
-			const sendEventWhenReady = () => {
-				const sendEventId =
-					contributionType === 'ONE_OFF'
-						? SendEventContributionAmountUpdate.SingleContribution
-						: SendEventContributionAmountUpdate.RecurringContribution;
-				const convertedValue = getContributionAnnualValue(
-					contributionType,
-					parseInt(amount),
-					sourceCurrency,
-				);
-				if (convertedValue) {
-					sendEvent(sendEventId, false, Math.round(convertedValue).toString());
-				}
-			};
+	void ifQmPermitted(() => {
+		const sendEventWhenReady = () => {
+			const sendEventId =
+				contributionType === 'ONE_OFF'
+					? SendEventContributionAmountUpdate.SingleContribution
+					: SendEventContributionAmountUpdate.RecurringContribution;
+			const convertedValue = getContributionAnnualValue(
+				contributionType,
+				parseInt(amount),
+				sourceCurrency,
+			);
+			if (convertedValue) {
+				sendEvent(sendEventId, false, Math.round(convertedValue).toString());
+			}
+		};
 
-			sendEventWhenReadyTrigger(sendEventWhenReady);
-		}
+		sendEventWhenReadyTrigger(sendEventWhenReady);
 	});
 }
 
 function sendEventContributionPaymentMethod(
 	paymentMethod: PaymentMethod | null,
 ): void {
 	if (paymentMethod) {
-		void canRunQuantumMetric().then((canRun) => {
-			if (canRun) {
-				const sendEventWhenReady = () => {
-					const sendEventId =
-						SendEventContributionPaymentMethodUpdate.PaymentMethod;
-					sendEvent(sendEventId, false, paymentMethod.toString());
-				};
-
-				sendEventWhenReadyTrigger(sendEventWhenReady);
-			}
+		void ifQmPermitted(() => {
+			const sendEventWhenReady = () => {
+				const sendEventId =
+					SendEventContributionPaymentMethodUpdate.PaymentMethod;
+				sendEvent(sendEventId, false, paymentMethod.toString());
+			};
+
+			sendEventWhenReadyTrigger(sendEventWhenReady);
 		});
 	}
 }
 
+function sendEventConversionPaymentMethod(paymentMethod: PaymentMethod): void {
+	void ifQmPermitted(() => {
+		sendEvent(
+			SendEventContributionPaymentMethodUpdate.PaymentMethodAtConversion,
+			false,
+			paymentMethod.toString(),
+		);
```

Comment:
Do you not need to use `sendEventWhenReadyTrigger` here? I guess the thinking is that since it’s called late enough in the process, we can assume quantum metric is ready. Is there any harm in using the check function anyway, just in case?